File: TheSwitcher_ObjC/TheSwitcher_ObjC/ViewControllers/ListViewController.m
//
//  ViewController.m
//  TheSwitcher_ObjC
//
//  Created by Fábio Maciel de Sousa on 10/02/21.
//

#import "ListViewController.h"

@interface ListViewController ()
@property DivisionsDataSource *dataSource;
@end

@implementation ListViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    NSArray *divisions = @[@"Kitchen",@"Living room",@"Master bedroom",@"Guest's bedroom"];
    self.dataSource = [[DivisionsDataSource alloc] initWith: divisions];
    self.tableView.dataSource = self.dataSource;
    self.tableView.tableFooterView = [[UIView alloc] init];
}

- (void)viewWillAppear:(BOOL)animated {
    NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
    if (indexPath) {
        [self.tableView deselectRowAtIndexPath: indexPath animated: true];
    }
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
    DetailViewController *detailViewController = [segue destinationViewController];
    DivisionsCell *cell = [self.tableView cellForRowAtIndexPath: indexPath];
    if (indexPath && detailViewController && cell) {
        bool isOn = [cell.lightSwitch isOn];
        detailViewController.status = isOn;
        detailViewController.division = cell.divisionLabel.text;
    }
}


@end

File: TheSwitcher_FabioMs/TheSwitcher_FabioMs/ListViewController.swift
//
//  ViewController.swift
//  TheSwitcher_FabioMs
//
//  Created by Fábio Maciel de Sousa on 10/02/21.
//

import UIKit

class ListViewController: UIViewController {
    @IBOutlet weak var tableView: UITableView!
    var dataSource: DivisionsDataSource?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        let divisions = [
            "Kitchen",
            "Living room",
            "Master bedroom",
            "Guest's bedroom"
        ]
        dataSource = DivisionsDataSource(With: divisions)
        tableView.dataSource = dataSource
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        if let indexPath = tableView.indexPathForSelectedRow {
            tableView.deselectRow(at: indexPath, animated: true)
        }
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        guard let indexPath = tableView.indexPathForSelectedRow,
              let detailViewController = segue.destination as? DetailViewController,
              let cell = tableView.cellForRow(at: indexPath) as? DivisionsCell else {
            return
        }
        detailViewController.division = cell.division
        detailViewController.lampState = cell.lightState
        
    }
}

